spec/behaves_spec.rb

```diff
@@ -63,6 +71,49 @@ def method_two; end
     end
   end
 
+  context 'private behaviors' do
+    before do
+      Interface.class_eval do
+        implements :foo
+        implements :bar, private: true
+      end
+    end
+
+    context 'when implemented' do
+      it 'raises no errors' do
+        expect do
+          Implementor.class_eval do
+            behaves_like Interface
+
+            def foo; end
+
+            private
+
+            def bar; end
+          end
+
+          Implementor.send(:check_for_unimplemented, Interface, :public)
+          Implementor.send(:check_for_unimplemented, Interface, :private)
+        end.not_to raise_error
+      end
+    end
+
+    context 'when unimplemented' do
+      it 'raises an error' do
+        expect do
+          Implementor.class_eval do
+            behaves_like Interface
+
+            def foo; end
+          end
+
+          Implementor.send(:check_for_unimplemented, Interface, :public)
+          Implementor.send(:check_for_unimplemented, Interface, :private)
+        end.to raise_error NotImplementedError, %r(private methods `bar`)
```

Comment:
It's an alternative regex syntax, usually used when you want to delimit with something other than `/` because your match has one in it:

```ruby
'foo/bar/baz'.match(/\/bar\//) # => #<MatchData "/bar/">
'foo/bar/baz'.match(%r(/bar/)) # => #<MatchData "/bar/">
```

I find it can be generally easier to parse visually though, so I tend to use it instead of `//` when special characters are involved. With my syntax highlighting in vim, the latter is a bit easier for me to read than the former:

```ruby
/`foo`/
%r(`foo`)
```

Also, syntactically, it works the same as `%w()`, so you can use delimiters other than `()` with it too:

```ruby
%r(foo) == %r"foo" # => true
%r(foo) == %r{foo} # => true
```

Reply:
Ah I see, that's pretty cool, TIL 😄